src/data-sources/plugins/GraphQL.ts

```diff
@@ -0,0 +1,74 @@
+import * as request from 'xhr-request';
+
+import { DataSourcePlugin, IOptions } from './DataSourcePlugin';
+import GraphQLConnection from '../connections/graphql';
+import { DataSourceConnector } from '../DataSourceConnector';
+
+let connectionType = new GraphQLConnection();
+
+interface IGraphQLParams {
+  query: string;
+  variables: Object;
+}
+
+export default class GraphQL extends DataSourcePlugin<IGraphQLParams> {
+  type = 'GraphQL';
+  defaultProperty = 'data';
+  connectionType = connectionType.type;
+
+  constructor(options: IOptions<IGraphQLParams>, connections: IDict<IStringDictionary>) {
+    super(options, connections);
+    this.validateParams(this._props.params);
+  }
+
+  updateDependencies(dependencies: any) {
+    // Ensure dependencies exist
+    const isAnyDependencyMissing = Object.keys(this.getDependencies()).some(key => dependencies[key] == null);
+    if (isAnyDependencyMissing) {
+      return dispatch => dispatch();
+    }
+
+    // Validate connection
+    const connection = this.getConnection();
+    const { serviceUrl } = connection;
+    if (!connection || !serviceUrl) {
+      return dispatch => dispatch();
+    }
+
+    const params = this.getParams();
+    const query = params.query;
```

Comment:
The suggested code doesn't compile, but I managed to add default values in e21d714.